src/libraries/System.Text.Json/src/System/Text/Json/Extensions/Utf8JsonReaderExtensions.cs

```diff
@@ -0,0 +1,304 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Buffers.Text;
+using System.Diagnostics;
+using System.Globalization;
+
+namespace System.Text.Json.Extensions
+{
+    internal static class Utf8JsonReaderExtensions
```

Comment:
As mentioned in https://github.com/dotnet/runtime/pull/114863#discussion_r2055572476, the read/write methods must not be directly called by `JsonValue` converters. They must resolve and call any registered `JsonConverter<TPrimitive>`.